forum_leaderboard/static/js/src/forum_leaderboard_studio.js

```diff
@@ -0,0 +1,21 @@
+/* Javascript for ForumLeaderboardXBlock. */
+function ForumLeaderboardStudioXBlock(runtime, element) {
+    $('.save-button', element).bind('click', function() {
+        var handlerUrl = runtime.handlerUrl(element, 'studio_submit');
+        var data = {
+            discussion_id: $(element).find('input[name=discussion_id]').val(),
+            count: $(element).find('input[name=count]').val()
+        };
+        $.post(handlerUrl, JSON.stringify(data)).success(function(response) {
+            if (response['success']) {
+                window.location.reload(false);
```

Comment:
Almost all the built in modules just reload in-place without reloading the whole page - but they use the standard studio editor. I'm not sure if there is a way for pure xblocks to do that yet. If not, there will be when they improve editing of pure xblock settings ([was partially implemented at the last hackathon](https://github.com/edx/edx-platform/pull/6000#) - as I recally they coded it so pure xblocks don't need custom edit page HTML and instead you just define the settings and studio creates the UI for you).